onnxruntime/core/providers/qnn/qnn_execution_provider.cc

```diff
@@ -274,28 +336,61 @@ Status QNNExecutionProvider::Compile(const std::vector<FusedNodeAndGraph>& fused
     LOGS(logger, VERBOSE) << "fused node name: " << fused_node.Name();
     qnn_models_.emplace(fused_node.Name(), std::move(qnn_model));
 
-    NodeComputeInfo compute_info;
-    compute_info.create_state_func = [&](ComputeContext* context, FunctionState* state) {
-      LOGS(logger, VERBOSE) << "compute_info.create_state_func context->node_name: " << context->node_name;
-      *state = qnn_models_[context->node_name].get();
-      return 0;
-    };
+    ORT_RETURN_IF_ERROR(CreateComputeFunc(node_compute_funcs, logger));
+  }
+  return Status::OK();
+}
 
-    compute_info.release_state_func = [](FunctionState state) {
-      // the 'state' is a qnn::QnnModel managed by unique_ptr
-      ORT_UNUSED_PARAMETER(state);
-    };
+bool QNNExecutionProvider::IsContextCacheFileExists(const onnxruntime::PathString& model_pathstring,
+                                                    onnxruntime::PathString& context_cache_pathstring) const {
+  // Use user provided context cache file path if exist, otherwise try model_file.onnx.bin by default
+  if (!context_cache_path_.empty()) {
+    context_cache_pathstring = ToPathString(context_cache_path_);
+  } else {
+    context_cache_pathstring = model_pathstring + ToPathString(".bin");
+  }
+  std::fstream context_cache_fs(context_cache_pathstring.c_str());
+  bool context_cache_file_exist = context_cache_fs.good() ? true : false;
```

Comment:
May be simpler to use `std::filesystem::exists()`: https://en.cppreference.com/w/cpp/filesystem/exists